components/ai_chat/core/browser/engine/conversation_api_client.cc

```diff
@@ -114,6 +114,35 @@ mojom::ConversationEntryEventPtr ParseResponseEvent(
     }
     return mojom::ConversationEntryEvent::NewSearchQueriesEvent(
         std::move(event));
+  } else if (*type == "webSources") {
+    const base::Value::List* sources = response_event.FindList("sources");
+    if (!sources) {
+      return nullptr;
+    }
+    auto event = mojom::WebSourcesEvent::New();
+    for (auto& item : *sources) {
+      if (!item.is_dict()) {
+        continue;
+      }
+      const base::Value::Dict& source = item.GetDict();
+      const std::string* title = source.FindString("title");
+      const std::string* url = source.FindString("url");
+      const std::string* favicon_url = source.FindString("favicon");
+      if (!title || !url || !favicon_url) {
+        DVLOG(2) << "Missing required fields in web source event: "
+                 << item.DebugString();
+        continue;
+      }
+      GURL item_url(*url);
+      GURL item_favicon_url(*favicon_url);
+      if (!item_url.is_valid() || !item_favicon_url.is_valid()) {
```

Comment:
i thought we were only loading favicons from imgs.search.brave.com? can we add that as a host check before loading the image?

Reply:
Yes I suppose we have to move the restriction to here now if we still want it, since the fetch is happening more generically and can't be enforced at the point of download.

Reply:
let's keep the restriction for now

Reply:
done